### PR TITLE
Update frontend.h to current master

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
The extern(C++) interface changed after the PR passed the CI

### CODE REVIEW

**Review by wilzbach — APPROVED** [review]

**Comment by MoonlightSentinel** [comment]
To avoid such errors in the future: Could CircleCI (and maybe other CI's) deprecate older jobs when the target branch is modifed?

**Comment by wilzbach** [comment]
> To avoid such errors in the future: Could CircleCI (and maybe other CI's) deprecate older jobs when the target branch is modifed?

The platforms don't support this AFAICT, but in theory we could do this automatically via dlang-bot ourselves. However, this will lead to a explosion in resource use (as now every PR merged into master, will trigger X new builds).

Other projects deal with this by requiring the PR to be up-to-date with the target before a merge can happen (GitHub itself has such a check) and then rebases (and thus rebuilds) would be triggered if the PR is in a merge queue.

**Comment by MoonlightSentinel** [comment]
> The platforms don't support this AFAICT, but in theory we could do this automatically via dlang-bot ourselves. However, this will lead to a explosion in resource use (as now every PR merged into master, will trigger X new builds).

Not ideal, but perhaps reasonable for one of the fast machines (to have some security)

> Other projects deal with this by requiring the PR to be up-to-date with the target before a merge can happen (GitHub itself has such a check) and then rebases (and thus rebuilds) would be triggered if the PR is in a merge queue.

That workw as well but is rather cumbersome for contributers (it might require multiple rebases for a single PR).
Maybe we could automate this via dlang-bot, e.g. extend `auto-merge` to `auto-rebase-merge`?

**Comment by wilzbach** [comment]
Well both paths are possible, but a bit of work to configure properly, but I agree this would be important to have before we fully get rid of the auto-tester.
As I won't have time to work on this for at least the foreseeable future, I don't plan to remove the further builds from the auto-tester.

Should we temporarily add the frontend header test target to the auto-tester-build  in `src/posix.mak`.

**Comment by MoonlightSentinel** [comment]
Good idea, added the check to the auto-tester (linux only) for now.

**Review by wilzbach — APPROVED** [review]

**Comment by MoonlightSentinel** [comment]
Moved it into the `unittest` target because it requires druntime

**Comment by MoonlightSentinel** [comment]
Limited to linux-64 because the linux-32 header uses different types.

File: src/dmd/frontend.h [hunk 1]
```diff
@@ -14187,18 +14190,30 @@ extern void verror(const Loc& loc, const char* format, va_list ap, const char* p
 
 extern void verrorSupplemental(const Loc& loc, const char* format, va_list ap);
 
+extern void _verrorSupplemental(const Loc& loc, const char* format, va_list ap);
```
Comment (wilzbach) [inline src/dmd/frontend.h:14193]: BTW do we really need to expose these wrapper functions to the outside? Shouldn't they be private and extern(D)?

Comment (MoonlightSentinel) [reply src/dmd/frontend.h:14193]: Not necessarily, seems like it's probably better to remove them from the C++ interface-

Comment (MoonlightSentinel) [reply src/dmd/frontend.h]: Done

**Pull request merged**